python/MooseDocs/extensions/content.py

```diff
@@ -281,3 +338,63 @@ def createHTML(self, parent, token, page):
 
     def createLatex(self, parent, token, page):
         return None
+
+class RenderContentOutline(components.RenderComponent):
+    def createHTML(self, parent, token, page):
+        if token['location'] is not None:
+            func = lambda p: p.local.startswith(token['location']) and isinstance(p, pages.Source)
+            nodes = self.translator.findPages(func)
+        elif token['pages']: # should I even check? It would have to be pages option at this point
```

Comment:
Doing the check is fine, but you should add an "else" that produces an error.

Reply:
We want to do this even thought I already checked for this conflict and gave an error in the command itself? [Here](https://github.com/idaholab/moose/blob/09c75bd44a2a602e3c832dd6c2e4b4c2e5a424c9/python/MooseDocs/extensions/content.py#L184-L190).

Reply:
Yes, some other user-create extension could create the token and do it wrong.